packages/cells/src/widget.ts

```diff
@@ -364,6 +364,14 @@ export class Cell extends Widget {
     this.editor.focus();
   }
 
+  /**
+   * Handle `fit-request` messages.
+   */
+  protected onFitRequest(msg: Message): void {
+    // need this for for when a theme changes font size
+    this.editor.refresh();
```

Comment:
We have run into performance problems in calling `refresh()` too much in the past. When exactly will `onFitRequest` be called and have we confirmed this doesn't degrade the performance?

Reply:
I should note the performance issue is when large notebooks are resized in the main work area area when all the cells have their CM refreshed.

Reply:
As I understand it, fit is only called by layouts and by us when we change the theme.  The layouts only call it when attaching or changing orientation (for box layout).

Reply:
Just to be safe I reviewed the relevant bits of the Phosphor and jlab code bases, and I tested the above `onFitRequest` again with a detailed logging function. 

- The only places in the jlab code that explicitly implement `fit` listeners are the terminal widget and the contents of this pull request. The only explicit call to `fit()` is in `thememanager`.

- Whenever a theme change occurs, the above `onFitRequest` fires exactly once for each extant cell. Otherwise it never gets called. I tried a bunch of different GUI interactions, but no amount of scrolling, resizing, dropping, or dragging triggered this function.

Reply:
OK great, thanks for looking into this. And Hi @blink1073 :-)